src/scala/org/pantsbuild/zinc/compiler/OutputUtils.scala

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2012 Typesafe, Inc. <http://www.typesafe.com>
+ */
+
+package org.pantsbuild.zinc.compiler
+
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{FileVisitResult, Files, Path, Paths, SimpleFileVisitor}
+import java.util.jar.{JarEntry, JarOutputStream}
+import scala.collection.mutable
+import sbt.util.Logger
+
+object OutputUtils {
+
+  /**
+   * Jar the contents of output classes (settings.classesDirectory) and copy to settings.outputJar
+   */
+  def createClassesJar(settings: Settings, log: Logger) = {
+    val classesDirectory = settings.classesDirectory
+    val sorted = new mutable.TreeSet[Path]()
+
+    val fileSortVisitor = new SimpleFileVisitor[Path]() {
+      override def preVisitDirectory(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        sorted.add(path)
+        FileVisitResult.CONTINUE
+      }
+      override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        sorted.add(path)
+        FileVisitResult.CONTINUE
+      }
+    }
+    // Sort the contents of the classesDirectory in lexicographic order for
+    // deterministic jar creation
+    Files.walkFileTree(classesDirectory.toPath, fileSortVisitor)
+
+    val jarPath = Paths.get(classesDirectory.toString, settings.outputJar.toString)
```

Comment:
It looks like this will make this a relative path under the classesDirectory?

Both of these should probably be (assumed to be) absolute, so that they don't need to be located within one another.

Reply:
^ as mentioned below, I think that this is still a thing... the test is currently assuming this behaviour.